### PR TITLE
BUG: Series.replace(method='pad') with EA dtypes

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/_mixins.py [hunk 1]
```diff
@@ -246,6 +246,14 @@ def __getitem__(
         result = self._from_backing_data(result)
         return result
 
+    def _pad_mask_inplace(
```
Comment (jreback) [inline pandas/core/arrays/_mixins.py:249]: why is this restricted to pad?

File: pandas/core/arrays/base.py [hunk 2]
```diff
@@ -1436,6 +1436,24 @@ def _where(
         result[~mask] = val
         return result
 
+    def _pad_mask_inplace(
```
Comment (jreback) [inline pandas/core/arrays/base.py:1439]: same shouldn't this be general? (e.g. the name pad is misleading here), mayb ejust _fillna_mask_inplace?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/base.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/arrays/base.py:1439]: _fill_mask_inplace?

File: pandas/core/arrays/_mixins.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/_mixins.py:249]: can be backfill too, will update name

**Comment by jreback** [comment]
I guess technically a user facing change? if so can you add a note

**Review by jreback — APPROVED** [review]

**Pull request merged**